codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/Protocol.kt

```diff
@@ -90,6 +90,13 @@ interface Protocol {
      * protocol.
      */
     fun serverRouterRuntimeConstructor(): String
+
+    /**
+     * In some protocols, such as RestJson,
+     * when there is no modeled body input, content type must not be set and the body must be empty
+     * Returns a writable to perform this check
+     */
+    fun serverContentTypeCheckNoModeledInput(): Boolean = false
```

Comment:
I don't want to block approval on this, but now that the prerequisite work has been done in #1698, it would be better to have this in the `ServerProtocol` implementations instead.

Reply:
After this is merged I'll move this in https://github.com/awslabs/smithy-rs/pull/1731